odxtools/compumethods/linearsegment.py

```diff
@@ -0,0 +1,176 @@
+    linear interpolation. This class centralizes the required
+    parameters for a single such segment. (These parameters are
+    extracted from the respective compu method's
+    COMPU-INTERNAL-TO-PHYS objects. We do it this way because the
+    internal-to-phys are rather clunky to work with and feature a lot
+    of irrelevant information.)
+    """
+    offset: float
+    factor: float
+    denominator: float
+    internal_lower_limit: Optional[Limit]
+    internal_upper_limit: Optional[Limit]
+
+    internal_type: DataType
+    physical_type: DataType
+
+    @staticmethod
+    def from_compu_scale(scale: CompuScale, *, internal_type: DataType,
+                         physical_type: DataType) -> "LinearSegment":
+        coeffs = odxrequire(scale.compu_rational_coeffs)
+
+        offset = coeffs.numerators[0]
+        factor = coeffs.numerators[1]
+
+        denominator = 1.0
+        if len(coeffs.denominators) > 0:
+            denominator = coeffs.denominators[0]
+
+        internal_lower_limit = scale.lower_limit
+        internal_upper_limit = scale.upper_limit
+
+        return LinearSegment(
+            offset=offset,
+            factor=factor,
+            denominator=denominator,
+            internal_lower_limit=internal_lower_limit,
+            internal_upper_limit=internal_upper_limit,
+            internal_type=internal_type,
+            physical_type=physical_type)
+
+    @property
+    def physical_lower_limit(self) -> Optional[Limit]:
+        return self._physical_lower_limit
+
+    @property
+    def physical_upper_limit(self) -> Optional[Limit]:
+        return self._physical_upper_limit
+
+    def __post_init__(self) -> None:
+        self.__compute_physical_limits__()
+
+    def convert_internal_to_physical(self, internal_value: AtomicOdxType) -> Union[float, int]:
+        if not isinstance(internal_value, (int, float)):
+            odxraise(f"Internal values of linear compumethods must "
+                     f"either be int or float (is: {type(internal_value).__name__})")
+
+        result = (self.offset + self.factor * internal_value) / self.denominator
+
+        if self.physical_type in [
+                DataType.A_INT32,
+                DataType.A_UINT32,
+        ]:
+            result = round(result)
+
+        return result
+
+    def convert_physical_to_internal(self, physical_value: AtomicOdxType) -> Union[float, int]:
+        if not isinstance(physical_value, (int, float)):
+            odxraise(f"Physical values of linear compumethods must "
+                     f"either be int or float (is: {type(physical_value).__name__})")
+
+        result = (physical_value * self.denominator - self.offset) / self.factor
+
+        if self.internal_type in [
+                DataType.A_INT32,
+                DataType.A_UINT32,
+        ]:
+            result = round(result)
+
+        return result
+
+    def __compute_physical_limits__(self) -> None:
```

Comment:
functions names that end and start with double underscore are reserved for python itself

Reply:
Aren't they just private methods, i.e. methods that cannot be called outside of the class?

Reply:
You're right, I looked this up: PEP 8 states that private methods use to leading underscores while leading and trailing underscores are reserved for magic functions. fixed: https://github.com/mercedes-benz/odxtools/pull/306/commits/58ea6317eeb654b607c5942462ba37bfa96ba503 (I wonder why `mypy` and `ruff` do not complain about this...)